### ISSUE TITLE
Met different access of __xfer_bufptrs while setting GoogleTest under CppUTest

### ISSUE THREAD

**Issue by gary1994613** [cpputest/cpputest#1652]
I tried to set up GoogleTest under CppUTest framework. I followed the description from the manual (https://cpputest.github.io/manual.html#gtest), but it failed to compile even for only the easiest TEST_GROUP HelloWorld in the example directory in CppUTest. I got stuck in the error:

> compiling AllTests.cpp
In file included from /usr/local/include/gtest/internal/gtest-port.h:271:0,
                 from /usr/local/include/gtest/internal/gtest-internal.h:40,
                 from /usr/local/include/gtest/gtest.h:62,
                 from ../include/CppUTestExt/GTest.h:37,
                 from ../include/CppUTestExt/GTestConvertor.h:95,
                 from AllTests/AllTests.cpp:34:
/usr/include/c++/7/sstream:300:7: error: ‘struct std::__cxx11::basic_stringbuf<_CharT, _Traits, _Alloc>::__xfer_bufptrs’ redeclared with different access
       struct __xfer_bufptrs
       ^~~~~~
compilation terminated due to -Wfatal-errors.

After little bit of exploring about the problems, I guess that this problem seems to be caused by the #define access hack in the include/CppUTestExt/GTestConvertor.h: 92
`#define private public
#define protected public
`
which makes the __xfer_bufptrs public but it is defined as private in class basic_stringbuf. I would like to know how should I handle this problem? 

Some refs:
- https://gcc.gnu.org/bugzilla/show_bug.cgi?id=65899
- https://stackoverflow.com/questions/47839718/sstream-redeclared-with-public-access-compiler-error

**Comment by basvodde** [comment]
Thanks. I do not know if CppUTest works with the latest version of gtest, so that might be the problem. Which GTest version did you use here ?

**Comment by gary1994613** [comment]
I followed the instructions from here: https://github.com/google/googletest/tree/main/googletest

> git clone https://github.com/google/googletest.git -b release-1.12.1
cd googletest        # Main directory of the cloned repository.
mkdir build          # Create a directory to hold the build output.
cd build
cmake ..             # Generate native build scripts for GoogleTest.

**Comment by basvodde** [comment]
Yes, at the moment, we've only tested CppUTest with max gtest 1.8. @thetic was checking if he can also get other version to work (and add it to the CppUTest automated build) but I suspect it might still need some work.

**Comment by thetic** [comment]
I was able to work around this by building with C++ 98. I'm starting to doubt that this can be made to work reliably across gtest and C++ versions.
